### ISSUE TITLE
Cannot use optional type

### ISSUE THREAD

**Issue by jd-carroll** [typescript-eslint/typescript-eslint#1581]
**What code were you trying to parse?**

```ts
// index.ts
export const myFunction = () => {
  let myVar?: string = undefined;
  // ...
};
```

**What did you expect to happen?**
The variable `myVar` to be define with the type `string | undefined` with the initial value of `undefined`.

**What actually happened?**
I receive a parsing error:
```
Parsing error: ',' expected. eslint
```
Note: The same happens even if I declare the variable like `let accountId?: string;`

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.3.1` |
| `TypeScript`                | `3.6.3` |
| `ESLint`                    | `6.4.0` |
| `node`                      | `10.17.0` |
| `npm`                       | `6.11.3` |

**Comment by armano2** [comment]
This syntax is not allowed in typescript [Playground Link](https://www.typescriptlang.org/play/?ssl=15&ssc=1&pln=1&pc=1#code/PTAEEsDsBMFMA8B0AXAzgWAFAIA4HsAnZUAYz0lWIFsBPAMQFdITlxzQBeUACgEpOAfKADeWUKAA2sajQBqAQwIB+AFyhKBKAHNOoJnABmUWNADcY0CFCIbWAL6mgA)

--------------

You have to use it like this:
```ts
// index.ts
export const myFunction = () => {
  let myVar: string | undefined = undefined;
  // ...
};
```
